### DISCUSSION TITLE
Proper way to write a function inside `<script setup>` ?

### DISCUSSION THREAD

**Discussion by telisjok** [nuxt/nuxt#18083] [Questions] [answered]
What is the proper way to write a function inside the `<script setup>` ?

Having this element to call a function when it gets clicked.
```
<div class="mobile-menu" @click="openMobileMenu('mobileMenu')">
  <img src="/assets/icons/icons-action-menu.svg" alt="user-icon" class="w-5 h-5" />
</div>
```

Is this the correct way to do this?
```
<script lang="ts" setup>
const openMobileMenu = (value: string) => {
  console.log(value);
}
</script>
```

This does nothing. And if I try to `{{ openMobileMenu }}` inside my `<template>` I will get `function(value) { console.log(value); }`

Before I change my code to `<script lang="ts" setup>` I had it like as shown below
```
<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: "Header",
  setup() {

    const openMobileMenu = (value: string) => {
      console.log(value);
    }

    return {
      openMobileMenu
    }
  }
})
</script>
```
and it was working fine.

**Comment by psycongaroo** [comment, marked as answer]
```
<script setup>
  function openMobileMenu(value) {
     console.log(value);
  }
</script>
```

**Reply by telisjok** [reply]
I tried that already and it doesn't do anything.

**Reply by psycongaroo** [reply]
> I tried that already and it doesn't do anything.

https://stackblitz.com/edit/github-y3vmo9-akp9a1?file=app.vue

It works

**Reply by telisjok** [reply]
Well, i'm pulling my hair here cause it won't work for me :/
